docs/intro/overview.rst

```diff
@@ -13,172 +13,83 @@ precisely, `web scraping`_), it can also be used to extract data using APIs
+and then called the default callback method ``parse`` passing the response
+object as an argument.
 
-Finally, here's the spider code::
+In the ``parse`` callback, we scrape the links to the questions and
+yield a few more requests to be processed, registering for them
+the method ``parse_question`` as the callback to be called when the
+requests are complete.
 
-    from scrapy.contrib.spiders import CrawlSpider, Rule
-    from scrapy.contrib.linkextractors import LinkExtractor
+Finally, the ``parse_question`` callback scrapes the question data
+for each page yielding a dict, which Scrapy then collects and
+writes to a JSON file as requested in the command line.
 
-    class MininovaSpider(CrawlSpider):
+.. note::
 
-        name = 'mininova'
-        allowed_domains = ['mininova.org']
-        start_urls = ['http://www.mininova.org/today']
-        rules = [Rule(LinkExtractor(allow=['/tor/\d+']), 'parse_torrent')]
+    This is using :ref:`feed exports <topics-feed-exports>` to generate the
+    JSON file, you can easily change the export format (XML or CSV, for example) or the
+    storage backend (FTP or `Amazon S3`_, for example).  You can also write an
+    :ref:`item pipeline <topics-item-pipeline>` to store the items in a database.
 
-        def parse_torrent(self, response):
-            torrent = TorrentItem()
-            torrent['url'] = response.url
-            torrent['name'] = response.xpath("//h1/text()").extract()
-            torrent['description'] = response.xpath("//div[@id='description']").extract()
-            torrent['size'] = response.xpath("//div[@id='info-left']/p[2]/text()[2]").extract()
-            return torrent
-
-The ``TorrentItem`` class is :ref:`defined above <intro-overview-item>`.
-
-Run the spider to extract the data
-==================================
-
-Finally, we'll run the spider to crawl the site and output the file
-``scraped_data.json`` with the scraped data in JSON format::
-
-    scrapy crawl mininova -o scraped_data.json
-
-This uses :ref:`feed exports <topics-feed-exports>` to generate the JSON file.
-You can easily change the export format (XML or CSV, for example) or the
-storage backend (FTP or `Amazon S3`_, for example).
-
-You can also write an :ref:`item pipeline <topics-item-pipeline>` to store the
-items in a database very easily.
-
-Review scraped data
-===================
-
-If you check the ``scraped_data.json`` file after the process finishes, you'll
-see the scraped items there::
-
-    [{"url": "http://www.mininova.org/tor/2676093", "name": ["Darwin - The Evolution Of An Exhibition"], "description": ["Short documentary made for Plymouth ..."], "size": ["150.62 megabyte"]},
-    # ... other items ...
-    ]
-
-You'll notice that all field values (except for the ``url`` which was assigned
-directly) are actually lists. This is because the :ref:`selectors
-<topics-selectors>` return lists. You may want to store single values, or
```

Comment:
Removing .extract_first() can make examples shorter, so it can be good for an overview :) We can explain what to do next in the tutorial.